models/staging/edfi_3/stage/stg_ef3__student_special_education_program_associations.sql

```diff
@@ -1,25 +1,27 @@
-with base_stu_spec_ed as (
+with base_stu_programs as (
     select * from {{ ref('base_ef3__student_special_education_program_associations') }}
     where not is_deleted
 ),
+
 keyed as (
     select 
         {{ gen_skey('k_student') }}, 
         {{ gen_skey('k_student_xyear') }}, 
         {{ gen_skey('k_program') }},
         {{ edorg_ref(annualize=False) }},
-        base_stu_spec_ed.*
+        api_year as school_year,
+        base_stu_programs.*
         {{ extract_extension(model_name=this.name, flatten=True) }}
-    from base_stu_spec_ed
+
+    from base_stu_programs
 ),
+
 deduped as (
-    {{
-        dbt_utils.deduplicate(
-            relation='keyed',
-            partition_by='k_student, ed_org_id, k_program, spec_ed_program_begin_date',
-            order_by='pull_timestamp desc'
-        )
-    }}
+    {{ dbt_utils.deduplicate(
+        relation='keyed',
+        partition_by='k_student, k_program, program_enroll_begin_date, school_year',
```

Comment:
school_year is part of k_program right?

Reply:
i think we can remove school_year from the dedupe of all these stg models, but it also doesn't really hurt anything

Reply:
It can be removed, but I think it also helps to be listed explicitly, even if just for internal understanding of the grain. Then again though, `school_year` should be encapsulated in `k_student` as well.